auto_gptq/modeling/_utils.py

```diff
@@ -101,8 +102,95 @@ def make_quant(
             desc_act=desc_act,
             trainable=trainable,
             disable_exllama=disable_exllama,
+            use_qigen=use_qigen
         )
 
+def process_zeros_scales(zeros, scales, bits, out_features):
+    if zeros.dtype != torch.float32:
+        new_zeros = torch.zeros_like(scales).float().contiguous()
+        if bits == 4:
+            qinfer.unpack_zeros4(zeros, new_zeros, new_zeros.shape[0], new_zeros.shape[1])
+        elif bits == 2:
+            qinfer.unpack_zeros2(zeros, new_zeros, new_zeros.shape[0], new_zeros.shape[1])
+        elif bits == 3:
+            logger.info("Unpacking zeros for 3 bits")
+        new_scales = scales.contiguous()
+    else:
+        if scales.shape[1] != out_features:
+            new_scales = scales.transpose(0,1).contiguous()
+        else:
+            new_scales = scales.contiguous()
+        if zeros.shape[1] != out_features:
+            new_zeros = zeros.transpose(0,1).contiguous()
+        else:
+            new_zeros = zeros.contiguous()
+
+    return new_zeros, new_scales
+
+def process_zeros_scales(zeros, scales, bits, out_features):
```

Comment:
@qwopqwop200 is this function duplicated with the one above it? Apart from this, I think this pr is good to be merged.

Reply:
oh what is this I guess I made a mistake.

Reply:
Now it seems good, thank you!!